### ISSUE TITLE
Reqest: allow for a pipe at end of map do block

### ISSUE THREAD

**Issue by fredcallaway** [timholy/ProgressMeter.jl#177]
I often have a structures like this in my code
```
results = map(jobs) do job
    job => work(job)
end |> Dict
```

If I try to throw a @showprogress on this, I get an error " ArgumentError: Final argument to @showprogress must be a for loop, comprehension, map, or pmap".  This can of course be fixed with a temporary variable, but if it's an easy fix to parse the do block here correctly, it would be more convenient.  More importantly, it would prevent confusion from aa somewhat mysterious error message.

**Comment by timholy** [comment]
If that can be made to work without breaking existing functionality, I'd be happy to have it. Want to submit a PR?

**Comment by timholy** [comment]
Closed by #179. (Pro tip: had you written "fixes #177" or "closes #177" instead of "addresses #177", this would have happened automatically.)
